pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -56,14 +64,23 @@ func (e transientError) Error() string {
 
 // Handle processes config and creates a new deployment if necessary.
 func (c *DeploymentConfigController) Handle(config *deployapi.DeploymentConfig) error {
-	// Inspect a deployment configuration every time the controller reconciles it
-	details, existingDeployments, latestDeploymentExists, err := c.findDetails(config)
+	existingDeployments, err := c.deploymentClient.listDeploymentsForConfig(config.Namespace, config.Name)
 	if err != nil {
 		return err
 	}
-	config, err = c.updateDetails(config, details)
-	if err != nil {
-		return transientError(err.Error())
+
+	// Inspect a deployment configuration every time the controller reconciles it
+	if config.Details == nil ||
+		config.Details.LastMessageUpdatedTime.IsZero() ||
+		time.Now().After(config.Details.LastMessageUpdatedTime.Add(c.messageUpdatePeriod)) {
```

Comment:
make `now` a func field in the controller that defaults to `time.Now` for testability?